src/Jobs/NodeChannelSuscribeJob.cs

```diff
@@ -0,0 +1,187 @@
+                            SatsAmount = channelOpened.Capacity,
+                            Status = Channel.ChannelStatus.Open,
+                            IsAutomatedLiquidityEnabled = false,
+                            BtcCloseAddress = channelOpened.CloseAddress,
+                            FundingTx = fundingTxAndIndex[0],
+                            FundingTxOutputIndex = Convert.ToUInt32(fundingTxAndIndex[1]),
+                            CreatedByNodeGuard = false,
+                            CreationDatetime = DateTimeOffset.Now,
+                            UpdateDatetime = DateTimeOffset.Now,
+                        };
+
+                        var remoteNode = await _nodeRepository.GetByPubkey(channelOpened.RemotePubkey);
+                        if (remoteNode == null)
+                        {
+                            var foundNode = await _lightningService.GetNodeInfo(channelOpened.RemotePubkey);
+                            if (foundNode == null)
+                            {
+                                throw new Exception("Node info not found");
+                            }
+                            
+                            remoteNode = new Node()
+                            {
+                                Name = foundNode.Alias,
+                                PubKey = foundNode.PubKey,
+                            };
+                            var addNode = await _nodeRepository.AddAsync(remoteNode);
+                            if (!addNode.Item1)
+                            {
+                                throw new Exception(addNode.Item2);
+                            }
+                        }
+                        
+                        remoteNode = await _nodeRepository.GetByPubkey(channelOpened.RemotePubkey);
+                        channel.SourceNodeId = channelOpened.Initiator ? node.Id : remoteNode.Id;
+                        channel.DestinationNodeId = channelOpened.Initiator ? remoteNode.Id : node.Id;
+                        
+                        var channelExists = await _channelRepository.GetByChanId(channel.ChanId);
+                        if (channelExists == null)
+                        {
+                            var addChannel = await _channelRepository.AddAsync(channel);
+                            if (!addChannel.Item1)
+                            {
+                                throw new Exception(addChannel.Item2);
+                            }
+
+                            _logger.LogInformation("Channel with id: {ChannelId} added to the system", channel.Id);
+                        }
+                        else
+                        {
+                            _logger.LogInformation("Channel with id: {ChannelId} already exists in the system", channel.Id);
+                        }
+                        
+                    }
+
+                    if (channelEventUpdate.ClosedChannel != null)
```

Comment:
Perhaps the channelEventUpdate types should be a switch instead of different ifs in the code

Reply:
Yeah, probably. It was only two types of event that we wanted so thats why I did an if

Reply:
Done